core/src/main/java/com/graphhopper/routing/util/parsers/BikeCommonPriorityParser.java

```diff
@@ -200,13 +200,16 @@ void collect(ReaderWay way, double wayTypeSpeed, TreeMap<Double, PriorityCode> w
             PriorityCode pushingSectionPrio = SLIGHT_AVOID;
             if (way.hasTag("bicycle", "yes") || way.hasTag("bicycle", "permissive"))
                 pushingSectionPrio = PREFER;
-            if (isDesignated(way))
+            if (isDesignated(way) && (!way.hasTag("highway","steps")))
                 pushingSectionPrio = VERY_NICE;
             if (way.hasTag("foot", "yes")) {
                 pushingSectionPrio = pushingSectionPrio.worse();
                 if (way.hasTag("segregated", "yes"))
                     pushingSectionPrio = pushingSectionPrio.better();
             }
+            if (way.hasTag("highway","steps")) {
+                pushingSectionPrio = BAD;
+            }
```

Comment:
Good catch, but it is neccessay because if we know that it is a pushing section (this is checked in that part on line 199) we use order 100 in the `weightToPrioMap` and the order for the `avoidHighwayTags` map gets only 50!

I now initialize the `avoidHighwayTags` for steps also with  `BAD` just to avoid confusion. This would not be needed as explained above.